### PR TITLE
Fix possibilities for test by user's role #414

### PR DESCRIPTION
**Pull request by draganakovacevic**

### CODE REVIEW

File: web/src/main/java/eu/execom/testgenerator/web/validator/FieldsValueMatchValidator.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    }
+
+    @Override
+    public boolean isValid(Map<Long, Integer> map, ConstraintValidatorContext context) {
+        for(Map.Entry<Long, Integer> entry: map.entrySet()) {
+            if(entry.getValue() == null) {
+                return false;
+            }
+        }
+        return true;
```
Comment (stanislavgrujic) [inline web/src/main/java/eu/execom/testgenerator/web/validator/FieldsValueMatchValidator.java:21]: Please format the file properly.

File: web/src/main/java/eu/execom/testgenerator/web/validator/FieldsValueMatch.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import javax.validation.Payload;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Constraint(validatedBy = FieldsValueMatchValidator.class)
+@Target({ElementType.METHOD, ElementType.FIELD})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface FieldsValueMatch {
```
Comment (stanislavgrujic) [inline web/src/main/java/eu/execom/testgenerator/web/validator/FieldsValueMatch.java:13]: Naming for this interface and associated validator is not intuitive. Could you please rename it to "AllAnswersReviewed"

File: web/src/main/java/eu/execom/testgenerator/web/validator/FieldsValueMatchValidator.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package eu.execom.testgenerator.web.validator;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.util.Map;
+
+public class FieldsValueMatchValidator implements ConstraintValidator<FieldsValueMatch, Map<Long, Integer>> {
```
Comment (stanislavgrujic) [inline web/src/main/java/eu/execom/testgenerator/web/validator/FieldsValueMatchValidator.java:7]: Please rename to "AllAnswersReviewedValidator".

File: web/src/main/java/eu/execom/testgenerator/web/validator/FieldsValueMatch.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+    String message() default "Question must have awarded points!";
```
Comment (stanislavgrujic) [inline web/src/main/java/eu/execom/testgenerator/web/validator/FieldsValueMatch.java:15]: No need to add exclamation mark in your messages to the user - it looks as if you are yelling at them. :) Please rephrase to "Points must be awarded to all answers."

File: web/src/main/frontend/src/base/stores/AuthStore.js [hunk 5]
```diff
@@ -1,9 +1,10 @@
 import {observable, action} from 'mobx';
 import authApi from '../rest/authApi';
-import {AUTH_TOKEN_KEY} from '../../components/utility/httpUtil';
+import {AUTH_TOKEN_KEY, USER_IS_ADMIN} from '../../components/utility/httpUtil';
 
 class AuthStore {
-  @observable isAdmin = true;
+
```
Comment (aberonja) [inline web/src/main/frontend/src/base/stores/AuthStore.js:6]: Same comment as in #421

File: web/src/main/frontend/src/base/stores/AuthStore.js [hunk 6]
```diff
@@ -1,9 +1,10 @@
...
+  @observable isAdmin;
```
Comment (aberonja) [inline web/src/main/frontend/src/base/stores/AuthStore.js:8]: Same comment as in #421

File: web/src/main/frontend/src/components/DashboardTableBody.js [hunk 7]
```diff
@@ -32,27 +32,41 @@ class DashboardTableBody extends React.Component {
...
-              />
-              <button
-                className="dashboard__trash_icon"
-                onClick={this.props.confirmPopup.showPopup(this.rowDelete(row))}
-                title="Delete test..."
-              />
-            </td>
-          </React.Fragment>
-        )}
+
```
Comment (aberonja) [inline web/src/main/frontend/src/components/DashboardTableBody.js:35]: Same comment as in #421

File: web/src/main/frontend/src/components/DashboardTableBody.js [hunk 8]
```diff
@@ -32,27 +32,41 @@ class DashboardTableBody extends React.Component {
...
+        <React.Fragment>
+          {this.props.isAdmin === 'true' ?
```
Comment (aberonja) [inline web/src/main/frontend/src/components/DashboardTableBody.js:37]: Same comment as in #421

File: web/src/main/java/eu/execom/testgenerator/web/validator/AllAnswersReviewedValidator.java [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+    @Override
+    public void initialize(AllAnswersReviewed constraintAnnotation) {
+
+    }
+
+    @Override
+    public boolean isValid(Map<Long, Integer> map, ConstraintValidatorContext context) {
+      for(Map.Entry<Long, Intseger> entry: map.entrySet()) {
+        if(entry.getValue() == null) return false;
```
Comment (stanislavgrujic) [inline web/src/main/java/eu/execom/testgenerator/web/validator/AllAnswersReviewedValidator.java:17]: I know that this looks simpler, but it is a bad practice to leave out brackets. Please add them around the return statement..

**Pull request merged**